### DISCUSSION TITLE
HDR merge of multiple exposure canon cr3 images with Magick.NET

### DISCUSSION THREAD

**Discussion by zanker99** [dlemstra/Magick.NET#650] [Help]
I just started to use Magick.NET with the hope of being able to do a HDR merge of multiple exposure canon cr3 raw images. 

Installed Magick.Net-Q16-HDRI-AnyCPU (ver:  7.17.0.1) using nuget with visual studio 2017, C Sharp Application on windows 10 machine. 

I have a few questions: 

1. Thanks for building that support in for canon cr3 raw images (realize Magick.NET is one of the few libraries that support cr3 images currently). I am able to read the cr3 raw images and convert to tiff ( which blows up the ~35 MB cr3 to ~140 MB Tiff file size ). 

     Can I assume that the output is a demosaiced RGB 16-bit per channel image ? 

```
    using (var image = new MagickImage("rawFile.cr3", MagickFormat.Cr3))
    {

         image.Write("rawFile.Tiff", MagickFormat.Tiff); 

     }
```

2. Is there an inbuilt function in Magick.NET that helps HDR merge of multi exposure images ? If not are there any Magick.NET samples that can help me along this direction.

**Comment by dlemstra** [comment]
I am not fully aware of what is done inside the libraw library that we use. The output depends on the `Depth` of the image. If the depth is `>=32` and the values are not out of the quantum range (`0.0-655535.0`) the tiff image will use floating points per channel. Otherwise because you use the `Q16` version of the library you will get 16-bit per channel.

And if you want to merge multiple images you probably wat to take a look at the `Composite` method. I am not sure which `CompositeOperator` you would have to use in your situation though.
